text/0000-use-case-integration-best-practices.md

```diff
@@ -0,0 +1,177 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+There are multiple ways that a use case can integrate with Hathor, here is a basic architecture of a common integration.
+
+- The use case has a software that runs operations offchain and is used by its customer.
+- This software connects with a headless wallet, that executes operations in the Hathor network.
+- The headless wallet is connected to a full node, that belongs to the Hathor peer-to-peer network.
+
+<pre>
+┌─────────────┐      ┌──────────┐                     ┌─────────────┐        ┌─────────────┐
+│             │      │          │◀────────────────────│             │        │             │
+│  Hathor     │      │ Headless │                     │  Use Case   │        │  Customer   │
+│  Full Node  │◀────▶│  Wallet  │                     │  Software   │◀──────▶│             │
+│             │      │          │────────────────────▶│             │        │             │
+└─────────────┘      └──────────┘                     └─────────────┘        └─────────────┘
+</pre>
+
+## Run more than one node
+
+Even though the architecture described above works fine for a simple use case, we strongly recommend any use case to run more than one node, in order to increase the reliability of the operation.
+
+During normal network activity, any transaction will be propagated through all peers and they will get to the same consensus. However, there are some situations where this might not be true, e.g. if an attack is happening to a single peer. To improve the reliability of the operation at any moment, it's important that all use cases run more than one full node and they **can't be connected among them**, only to the rest of the network.
+
+Let's assume an exchange wants to run nodes to identify deposits in the Hathor network, so a recommended approach for the integration would be to run at least two full nodes (node1 and node2), which are not connected between them (node1 must have node2 in the blacklist and vice versa). In that architecture, if any deposit is identified in node1, then the exchange must check that it's also a valid transaction in node2 and in one of the public nodes. With this approach, if one of the nodes is compromised and with a different consensus/state, an alert must be raised, so we must investigate what happened.
+
+### Validate new transactions on more than one full node before accepting them
+
+When running more than one node, any new transaction must be accepted only after validating that it's a valid transaction in all the nodes.
+
+### Validate all your full nodes have the same best block
+
+It's also important to regularly check if all nodes have the same best block. If they have different best blocks, this might be a clue that one of the nodes could be a target of an attack because is running a separate fork of the blockchain.
```

Comment:
Suggestion: "Use cases should regularly check whether the best block is the same on all their full nodes. If full node have different best blocks, they might be under attack and should consider blocking deposits."

I feel we should improve this recommendation because a naive implementation that collects the best block from all full nodes and compare might fail due to a race condition. For example, a new block found during the data collection.

Reply:
Improved the sentence to suggest executing more validations because of the block propagation time difference.